mmv1/third_party/terraform/services/compute/data_source_google_compute_reservation_test.go

```diff
@@ -0,0 +1,194 @@
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+)
+
+func TestComputeReservationIdParsing(t *testing.T) {
+	cases := map[string]struct {
+		ImportId            string
+		ExpectedError       bool
+		ExpectedCanonicalId string
+		Config              *transport_tpg.Config
+	}{
+		"id is a full self link": {
+			ImportId:            "https://www.googleapis.com/compute/v1/projects/test-project/zones/us-west1-a/reservations/test-reservation",
+			ExpectedError:       false,
+			ExpectedCanonicalId: "projects/test-project/zones/us-west1-a/reservations/test-reservation",
+		},
+		"id is a partial self link": {
+			ImportId:            "projects/test-project/zones/us-west1-a/reservations/test-reservation",
+			ExpectedError:       false,
+			ExpectedCanonicalId: "projects/test-project/zones/us-west1-a/reservations/test-reservation",
+		},
+		"id is project/zone/reservation": {
+			ImportId:            "test-project/us-west1-a/test-reservation",
+			ExpectedError:       false,
+			ExpectedCanonicalId: "projects/test-project/zones/us-west1-a/reservations/test-reservation",
+		},
+		"id is zone/reservation": {
+			ImportId:            "us-west1-a/test-reservation",
+			ExpectedError:       false,
+			ExpectedCanonicalId: "projects/default-project/zones/us-west1-a/reservations/test-reservation",
+			Config:              &transport_tpg.Config{Project: "default-project"},
+		},
+		"id is reservation": {
+			ImportId:            "test-reservation",
+			ExpectedError:       false,
+			ExpectedCanonicalId: "projects/default-project/zones/us-west1-a/reservations/test-reservation",
+			Config:              &transport_tpg.Config{Project: "default-project", Zone: "us-west1-a"},
+		},
+		"id has invalid format": {
+			ImportId:      "i/n/v/a/l/i/d",
+			ExpectedError: true,
+		},
+	}
+
+	for tn, tc := range cases {
+		reservationId, err := compute.ParseComputeReservationId(tc.ImportId, tc.Config)
+
+		if tc.ExpectedError && err == nil {
+			t.Fatalf("bad: %s, expected an error", tn)
+		}
+
+		if err != nil {
+			if tc.ExpectedError {
+				continue
+			}
+			t.Fatalf("bad: %s, err: %#v", tn, err)
+		}
+
+		if reservationId.CanonicalId() != tc.ExpectedCanonicalId {
+			t.Fatalf("bad: %s, expected canonical id to be `%s` but is `%s`", tn, tc.ExpectedCanonicalId, reservationId.CanonicalId())
+		}
+	}
+}
+
+func TestAccDataSourceComputeReservation(t *testing.T) {
+	t.Parallel()
+
+	reservationName := fmt.Sprintf("tf-test-%s", acctest.RandString(t, 10))
+
+	rsName := "foobar"
+	rsFullName := fmt.Sprintf("google_compute_reservation.%s", rsName)
+	dsName := "my_reservation"
+	dsFullName := fmt.Sprintf("data.google_compute_reservation.%s", dsName)
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccCheckDataSourceComputeReservationDestroy(t, rsFullName),
```

Comment:
You can most likely reuse `testAccCheckComputeReservationDestroyProducer` here rather than defining a new one - that's the normal pattern.